src/libs/UnreadIndicatorUpdater/index.js

```diff
@@ -1,14 +1,26 @@
 import _ from 'underscore';
 import Onyx from 'react-native-onyx';
+import {InteractionManager} from 'react-native';
 import ONYXKEYS from '../../ONYXKEYS';
 import updateUnread from './updateUnread/index';
 import * as ReportUtils from '../ReportUtils';
 
+let previousUnreadCount = 0;
+
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.REPORT,
     waitForCollectionCallback: true,
     callback: (reportsFromOnyx) => {
-        const unreadReports = _.filter(reportsFromOnyx, ReportUtils.isUnread);
-        updateUnread(_.size(unreadReports));
+        if (!reportsFromOnyx) {
+            return;
+        }
+
+        InteractionManager.runAfterInteractions(() => {
+            const unreadReportsCount = _.filter(reportsFromOnyx, ReportUtils.isUnread).length || 0;
+            if (previousUnreadCount !== unreadReportsCount) {
+                previousUnreadCount = unreadReportsCount;
+                updateUnread(unreadReportsCount);
+            }
+        });
```

Comment:
Can you add a comment here explaining why we need this interaction manager here?

Reply:
Yes !